x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elasticsearch/ElasticsearchInternalService.java

```diff
@@ -562,6 +561,7 @@ private static CustomElandEmbeddingModel updateModelWithEmbeddingDetails(CustomE
             model.getServiceSettings().getNumThreads(),
             model.getServiceSettings().modelId(),
             model.getServiceSettings().getAdaptiveAllocationsSettings(),
+            model.getServiceSettings().getDeploymentId(),
```

Comment:
This is the fix. The rest of the changes are removing constructors or reducing the visibility of the ctors so that it is not possible to make this mistake